x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/huggingface/HuggingFaceServiceTests.java

```diff
@@ -86,18 +89,22 @@ public void testParseRequestConfig_CreatesAnEmbeddingsModel() throws IOException
                 new SetOnce<>(createWithEmptySettings(threadPool))
             )
         ) {
-            var model = service.parseRequestConfig(
+
+            ActionListener<Model> modelVerificationActionListener = ActionListener.<Model>wrap((model) -> {
+                assertThat(model, instanceOf(HuggingFaceEmbeddingsModel.class));
```

Comment:
Just curious, if you intentionally make one of the asserts fail, will the test fail? For example switch this line to

```
assertThat(model, instanceOf(String.class))
``

Typically the way I've tested code that needs an action listener is to pass in a `PlainActionFuture` and then call `.get(timeout, timeout units)` that way you can interact with the result instead of having to create `onResponse` and `onFailure` functions. If it would call `onFailure` it'll just fail the test.

Here's an example: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/openai/OpenAiServiceTests.java#L791

Reply:
I can confirm that doing that line replacement does cause the test to fail, as expected.

In retrospect that probably would have been an easier refactoring, but I don't think it should make any practical difference at that point to use an ActionFuture versus putting the testing in the ActionListener. Do you see any functional differences between these approaches? Or any other reason to go back and do this refactoring in that way?

Reply:
You can leave what you have. I just wanted to make sure that it would fail.